src/Microsoft.Azure.WebJobs.Host/TypeUtility.cs

```diff
@@ -98,6 +98,68 @@ internal static Type UnwrapTaskType(Type type)
             return null;
         }
 
+        // $$$
+        internal static TAttribute GetAttr<TAttribute>(ParameterInfo parameter) where TAttribute : Attribute
+        {
+            TAttribute attributeSource = parameter.GetCustomAttribute<TAttribute>(inherit: false);
+
+            // Walk hierachy.
+            // It's possible attribute is not on the parameter but is still on the something 
+            // higher up in the hiearchy
+            ICustomAttributeProvider provider = parameter;
+
+            while (provider != null)
```

Comment:
GetHierarchicalAttributeOrNull  doesn't handle base classes. We do leverage existing tests on the new code.  I can probably remove GetHierarchicalAttributeOrNull  so that there's only one copy of this.